src/store-api/src/region_request.rs

```diff
@@ -153,6 +153,10 @@ impl RegionRequest {
             region_request::Body::Drops(drops) => make_region_drops(drops),
             region_request::Body::Alters(alters) => make_region_alters(alters),
             region_request::Body::BulkInserts(bulk) => make_region_bulk_inserts(bulk),
+            region_request::Body::Sync(_) => UnexpectedSnafu {
+                reason: "Sync request should be handled separately by RegionServer::handle_sync_region_request",
```

Comment:
non-blocking: We may not expose the method name to the message. But we can leave a comment here.
```suggestion
                reason: "Sync request should be handled separately by RegionServer",
```